test/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java

```diff
@@ -325,49 +335,57 @@ public void testCrNocr() {
+		assertTrue(noCrArg+ " should come after "+DJAVA_HOME, posns.get(crArg) < posns.get(noCrArg));
 	}
 
 	/* test options in runnable jar files */
 	
 	public void testJarArguments() {
-		String cmdlineArg = "-DcmdlineArg";
-		String testJar = System.getProperty("org.openj9.test.vmArguments.testJar");
-		URL jarUrl = ClassLoader.getSystemClassLoader().getResource("vmargs.jar");
-		assertNotNull("vmargs.jar not found", jarUrl);
-		testJar = jarUrl.getFile();
-		if (isWindows() && testJar.startsWith("/")) {
-			testJar = testJar.substring(1); /* annoying habit of getResource() to put a / at the front of the path */
+		URL jarUrl = ClassLoader.getSystemClassLoader().getResource(vmargsJarFilename);
+		assertNotNull(vmargsJarFilename+" not found", jarUrl);
+		Path jarPath = null;
+		try {
+			jarPath = Paths.get(jarUrl.toURI()).toFile().toPath();
+			checkJarArgs(jarPath.toString());
+		} catch (URISyntaxException e) {
+			fail("cannot convert "+jarUrl.toString()+" to path", e);
 		}
-		ProcessBuilder pb = makeProcessBuilder(null, new String[] {cmdlineArg, "-jar", testJar}, null);
-		Map<String, String> env = pb.environment();
-		String javaToolOptionsArg = "-DjavaToolOptionsArg";
-		env.put(JAVA_TOOL_OPTIONS, javaToolOptionsArg );
-		String ibmJavaOptionsArg = "-DibmJavaOptionsArg";
-		env.put(IBM_JAVA_OPTIONS, ibmJavaOptionsArg );
-		ArrayList<String> actualArguments = runAndGetArgumentList(pb);
-		String fooBar = "-Dfoo=bar";
-		String longProp = "-Da.long.system.property=this is a long system property value to demonstrate long JVM arguments in the manifest file";
-		HashMap<String, Integer> posns = checkArguments(actualArguments, new String[] {javaToolOptionsArg, ibmJavaOptionsArg, cmdlineArg, fooBar,
-		longProp});
-		assertTrue("missing argument: "+fooBar, posns.containsKey(fooBar));
-		assertTrue("missing argument: "+longProp, posns.containsKey(longProp));
-		/* environment variables should come after implicit arguments */
-		int toolsPosn = posns.get(javaToolOptionsArg);
-		int ibmPropPosn = posns.get(ibmJavaOptionsArg);
-		int fooBarPropPosn = posns.get(fooBar);
-		int longPropPosn = posns.get(longProp);
-		Integer cmdlineArgPosn = posns.get(cmdlineArg);
-		assertTrue("Wrong order of properties", ibmPropPosn == toolsPosn + 1);
-		assertTrue("Wrong order of properties", fooBarPropPosn < ibmPropPosn);
-		assertTrue("Wrong order of properties", fooBarPropPosn < cmdlineArgPosn);
-		assertTrue("Wrong order of properties", longPropPosn == fooBarPropPosn + 1);
+	}
+
+	/**
+	 * Test getting VM arguments from a JAR file prefixed by a shell script.
+	 */
+	public void testExecutableJarArguments() {
+		final String HWVMARGS = "hwvmargs.jar";
+		URL jarUrl = ClassLoader.getSystemClassLoader().getResource(vmargsJarFilename);
+		final String pathString = HWVMARGS;
+		assertNotNull(pathString+" not found", jarUrl);		
+		Path jarPath = null;
+		try {
+			jarPath = Paths.get(jarUrl.toURI()).toFile().toPath();
+		} catch (URISyntaxException e) {
```

Comment:
Just let this throw.  The test framework will fail the test for you

Reply:
But I lose the context, i.e. the problematic URL.  That's why I catch and rethrow.